source/common/router/rds_impl.cc

```diff
@@ -119,5 +125,45 @@ void RdsRouteConfigProviderImpl::registerInitTarget(Init::Manager& init_manager)
   init_manager.registerTarget(*this);
 }
 
+HttpRouteManagerImpl::HttpRouteManagerImpl(Runtime::Loader& runtime, Event::Dispatcher& dispatcher,
+                                           Runtime::RandomGenerator& random,
+                                           const LocalInfo::LocalInfo& local_info,
+                                           ThreadLocal::SlotAllocator& tls)
+    : runtime_(runtime), dispatcher_(dispatcher), random_(random), local_info_(local_info),
+      tls_(tls) {}
+
+std::vector<Router::RouteConfigProviderSharedPtr> HttpRouteManagerImpl::routeConfigProviders() {
+  std::vector<Router::RouteConfigProviderSharedPtr> ret;
+  ret.reserve(route_config_providers_.size());
+  for (const auto& element : route_config_providers_) {
+    ret.push_back(element.second.lock());
```

Comment:
threading has nothing to do with it here. All of this is single threaded. I haven't looked at the code yet but assuming it's correct lock() should never return nullptr.